botnim/vector_store/vector_store_es.py

```diff
@@ -79,135 +81,78 @@ def parse_index_name(index_name: str) -> str:
-                    "metadata.title",
-                    "metadata.extracted_data.DocumentTitle",
-                    "metadata.extracted_data.DocumentTitle.keyword^3",  # Boost keyword matches
-                    "metadata.extracted_data.OfficialSource",
-                    "metadata.extracted_data.OfficialRoles.Role",
-                    "metadata.extracted_data.Description",
-                    "metadata.extracted_data.AdditionalKeywords",
-                    "metadata.extracted_data.Topics",
-                ],
-                "boost": 0.4,           # boost for text match vs vector match
-                "type": 'cross_fields',  # type of search: cross_fields, bool, simple, phrase, phrase_prefix
-                "operator": 'or',       # operator: or, and 
-            }
-        }
-        
-        # Enhanced priority for document title matches
-        # This creates several potential exact title matches by taking the first 1-3 words of the query
-        title_matches = []
-        words = query_text.split()
-        
-        # Try exact match with full query
-        title_matches.append({
-            "term": {
-                "metadata.extracted_data.DocumentTitle.keyword": {
-                    "value": query_text,
-                    "boost": 10.0
-                }
-            }
-        })
+    def _build_search_query(
+        self,
+        query_text: str,
+        search_mode: Optional[SearchModeConfig] = None
+    ) -> Dict[str, Any]:
+        """
+        Builds an Elasticsearch query based on the search mode configuration.
         
-        # Try with first 1-3 words (to match document titles like "חוק הכנסת")
-        if len(words) >= 2:
-            potential_title = " ".join(words[0:2])
-            title_matches.append({
-                "term": {
-                    "metadata.extracted_data.DocumentTitle.keyword": {
-                        "value": potential_title,
-                        "boost": 20.0  # Higher boost for shorter, exact title matches
-                    }
-                }
-            })
+        Args:
+            query_text: The search query string
+            search_mode: Optional search mode configuration
             
-        if len(words) >= 3:
-            potential_title = " ".join(words[0:3])
-            title_matches.append({
-                "term": {
-                    "metadata.extracted_data.DocumentTitle.keyword": {
-                        "value": potential_title,
-                        "boost": 15.0
-                    }
-                }
-            })
-        
-        # Add a prefix match as fallback
-        title_matches.append({
-            "prefix": {
-                "metadata.extracted_data.DocumentTitle.keyword": {
-                    "value": words[0] if words else "",
-                    "boost": 5.0
+        Returns:
+            Dict containing the Elasticsearch query
+        """
+        if not search_mode:
```

Comment:
the fallback mode does match the previous behavior in terms of functionality, just with a slightly different query structure that's more consistent with the new search mode implementation.